src/Microsoft.SqlTools.ServiceLayer/AutoParameterizaition/SqlParameterizer.cs

```diff
@@ -0,0 +1,128 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.SqlServer.TransactSql.ScriptDom;
+using System.Linq;
+using Microsoft.SqlTools.ServiceLayer.AutoParameterizaition.Exceptions;
+using System.Data.Common;
+using Microsoft.SqlTools.ServiceLayer.Workspace.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.AutoParameterizaition
+{
+    public static class SqlParameterizer
+    {
+        private const int maxStringLength = 300000; // Approximately 600 Kb
+        private static readonly IList<ScriptFileMarker> EmptyCodeSenseItemList = Enumerable.Empty<ScriptFileMarker>().ToList();
+
+        public static SqlScriptGenerator GetScriptGenerator()
+        {
+            return new Sql150ScriptGenerator();
+        }
+
+        public static TSqlParser GetTSqlParser(bool initialQuotedIdentifiers)
+        {
+            return new TSql150Parser(initialQuotedIdentifiers);
```

Comment:
So I remember leaving this comment in the original implementation in SSMS - but isn't this going to cause problems as time goes on and new syntax is added to newer versions? Using a hardcoded parser version like this means it'll fail to parse that until we go in and update this.

Can't you just fetch the latest parser version available and use that? That was as we update the parser lib it'll keep itself up to date

Reply:
That is a good point and something I'll ask Brian about. If there is a way to get the latest parser from DacFx then I will make use of it.

Reply:
I asked the DacFx folks and they told me that there is no method to get the latest version of the parser/scriptor currently.